pkg/storage/unified/client.go

```diff
@@ -153,3 +161,53 @@ func newResourceClient(conn *grpc.ClientConn, cfg *setting.Cfg, features feature
 	}
 	return resource.NewRemoteResourceClient(tracer, conn, clientCfgMapping(grpcutils.ReadGrpcClientConfig(cfg)), cfg.Env == setting.Dev)
 }
+
+// grpcConn creates a new gRPC connection to the provided address.
+func grpcConn(address string, reg prometheus.Registerer) (*grpc.ClientConn, error) {
+	// This works for now as the Provide function is only called once during startup.
+	// We might eventually want to tight this factory to a struct for more runtime control.
+	metrics := ClientMetrics{
```

Comment:
Curious why we even introduce `metrics` and `ClientMetrics`. Having `requestDuration` would be enough.

Reply:
There is a follow up PR adding more client metrics. Like retry count etc.